### ISSUE TITLE
`crossterm::style::Stylize` is imported instead of `ratatui::style::Stylize`

### ISSUE THREAD

**Issue by orhun** [ratatui/ratatui#1347]
## Description

I have a feeling that I shouldn't be the only one having this issue with my LSP...

## To Reproduce

I'm using Neovim with `rust-analyzer`, and every time I use the style shortcuts for Ratatui, such as `.yellow()`, the wrong trait gets imported.

We might want to consider **renaming** this trait to avoid name clashes. Alternatively, if anyone knows how to configure `rust-analyzer` to import the correct trait, that would be helpful.

## Expected behavior

Correct imports.

## Screenshots

![image](https://github.com/user-attachments/assets/006b2c00-5680-44de-98b7-db03924aedcc)

## Environment

Ratatui version: `0.28.1`
Backend: `crossterm`
Editor: Neovim
LSP: `rust-analyzer`

## Additional context

n/a

**Comment by joshka** [comment]
Yeah this is a pain. I've raised a similar issue on the r-a issue tracker before (https://github.com/rust-lang/rust-analyzer/issues/17652), and there's a need to do a few things there:
- log in the completion relevance functions to help diagnose these issues
- have some configuration to prevent certain imports. This should likely be available at both the editor and workspace level.

From a ratatui side, there's no change that would really fix this properly, as you will still get the same completions even if Stylize is renamed. There's nothing I can see that would be make it possible for r-a to disambiguate that choice automatically.

Perhaps at the language level there needs to be some attribute that would hint certain imports as preferred over others? That could be a useful approach.

**Comment by kdheepak** [comment]
In neovim there’s a couple of extensions that you can use to order completions:  

1. https://github.com/ryo33/nvim-cmp-rust
2. https://github.com/zjp-CN/nvim-cmp-lsp-rs

Here’s a link to my nvim configuration to see it in action: https://github.com/kdheepak/nvim/blob/186414fc27ed3c230bc5934dca3ce75677933efd/lua/kd/plugins/nvim-cmp.lua#L40

**Comment by joshka** [comment]
I was tempted to close this, but I'd suggest that would instead make a good FAQ entry.

How do I configure rust analyzer to favor ratatui imports over crossterm / owo_color imports when autocompleting?

**Comment by orhun** [comment]
> From a ratatui side, there's no change that would really fix this properly, as you will still get the same completions even if Stylize is renamed.

Ah, right. Because the function names will still be the same 💀 (`.red()` etc.)

> I was tempted to close this, but I'd suggest that would instead make a good FAQ entry.

Totally agree. I will try out kd's suggestions now and slap together a PR if it works.

**Comment by orhun** [comment]
I tried this minimal configuration but unfortunately it still doesn't work for me:

```lua
{
  "hrsh7th/nvim-cmp",
  dependencies = {
    {
      "zjp-CN/nvim-cmp-lsp-rs",
      ---@type cmp_lsp_rs.Opts
      opts = {
        -- Filter out import items starting with one of these prefixes.
        -- A prefix can be crate name, module name or anything an import
        -- path starts with, no matter it's complete or incomplete.
        -- Only literals are recognized: no regex matching.
        unwanted_prefix = { "ratatui::crossterm::style::Stylize", "crossterm", "Styled" },
      },
    },
  },
}
```

![image](https://github.com/user-attachments/assets/927d6ae2-45f2-49a8-a38b-23fcc5f837de)
